src/lib/Tests/Specification/VersionHasConflictTest.php

```diff
@@ -0,0 +1,124 @@
+                $this->createVersionInfo(VersionInfo::STATUS_ARCHIVED),
+            ]);
+
+        $versionHasConflict = new VersionHasConflict($contentServiceMock);
+
+        self::assertFalse($versionHasConflict->isSatisfiedBy($this->createVersionInfo(VersionInfo::STATUS_DRAFT, 5)));
+    }
+
+    public function testVersionWithStatusDraftAndVersionConflict()
+    {
+        /** @var ContentService|MockObject $contentServiceMock */
+        $contentServiceMock = $this->createMock(ContentService::class);
+        $contentServiceMock
+            ->method('loadVersions')
+            ->willReturn([
+                $this->createVersionInfo(VersionInfo::STATUS_PUBLISHED, 4),
+                $this->createVersionInfo(VersionInfo::STATUS_DRAFT, 3),
+                $this->createVersionInfo(VersionInfo::STATUS_ARCHIVED),
+            ]);
+
+        $versionHasConflict = new VersionHasConflict($contentServiceMock);
+
+        self::assertTrue($versionHasConflict->isSatisfiedBy($this->createVersionInfo(VersionInfo::STATUS_DRAFT, 2)));
+    }
+
+    /**
+     * @dataProvider versionWithStatusOtherThanDraftProvider
+     *
+     * @param VersionInfo $versionInfo
+     */
+    public function testVersionWithStatusOtherThanDraft(VersionInfo $versionInfo)
+    {
+        $this->expectException(BadStateException::class);
+        $this->expectExceptionMessage('Argument \'Version status\' has a bad state: status is not draft');
+
+        /** @var ContentService|MockObject $contentServiceMock */
+        $contentServiceMock = $this->createMock(ContentService::class);
+
+        $versionHasConflict = new VersionHasConflict($contentServiceMock);
+        $versionHasConflict->isSatisfiedBy($versionInfo);
+    }
+
+    public function versionWithStatusOtherThanDraftProvider(): array
+    {
+        return [
+            'archived_version' => [
+                $this->createVersionInfo(VersionInfo::STATUS_ARCHIVED),
+            ],
+            'published_version' => [
+                $this->createVersionInfo(VersionInfo::STATUS_PUBLISHED),
+            ],
+        ];
+    }
+
+    /**
+     * Returns VersionInfo.
+     *
+     * @param int $status
+     * @param int $versionNo
+     *
+     * @return VersionInfo
+     */
+    private function createVersionInfo(int $status, int $versionNo = 1): VersionInfo
+    {
+        $contentInfo = $this->createMock(ContentInfo::class);
+
+        return new class(['status' => $status, 'contentInfo' => $contentInfo, 'versionNo' => $versionNo]) extends VersionInfo {
+            /** @var string */
+            protected $contentInfo;
+
+            /**
+             * @param array $properties
+             */
+            public function __construct(array $properties)
+            {
+                parent::__construct($properties);
+                $this->contentInfo = $properties['contentInfo'];
+            }
+
+            public function getNames()
+            {
```

Comment:
There is no point in implementing an anonymous class which takes 20+ lines of code and half of the code are invalid methods (against the interface, I'd expect `getNames()` to at least return an empty array). Please stick to mocks, they looks 100 times better and doesn't require you to write invalid code.